components/form/builder/src/index.js

```diff
@@ -53,52 +53,56 @@ const FormBuilder = ({
     return nextFields
   }
 
-  const handlerChange = useCallback(async (id, value, extraParams) => {
-    const reducerWithRules = reducer(
-      rules,
-      formID,
-      baseAPIURL,
-      responseInterceptor,
-      requestInterceptor,
-      locale,
-      extraParams
-    )
-    const timerShowSpinner = setTimeout(
-      () => setStateShowSpinner(true),
-      FormBuilder.USER_MINIMAL_DELAY
-    )
-    const transformedValue = transformations(
-      id,
-      value,
-      pickFieldById(fields, id)
-    )
+  const handlerChange = useCallback(
+    async (id, value, extraParams, autocompletedField = false) => {
+      const reducerWithRules = reducer(
+        rules,
+        formID,
+        baseAPIURL,
+        responseInterceptor,
+        requestInterceptor,
+        locale,
+        extraParams
+      )
+      const timerShowSpinner = setTimeout(
+        () => setStateShowSpinner(true),
+        FormBuilder.USER_MINIMAL_DELAY
+      )
+      const transformedValue = transformations(
+        id,
+        value,
+        pickFieldById(fields, id)
+      )
+
+      const nextFields = changeField(id, transformedValue)
+      clearTimeout(timerShowSpinner)
+
+      const nextStateFields = await reducerWithRules(nextFields, {
+        type: RULES,
+        id
+      })
 
-    const nextFields = changeField(id, transformedValue)
-    clearTimeout(timerShowSpinner)
-
-    const nextStateFields = await reducerWithRules(nextFields, {
-      type: RULES,
-      id
-    })
-
-    clearTimeout(timerShowSpinner)
-    setStateFields(nextStateFields)
-    const nextStateFieldsObject = useNativeFieldType
-      ? fieldsToObjectNativeTypes(nextStateFields)
-      : fieldsToObject(nextStateFields)
-
-    const fieldsToUpdate = await onChange({
-      ...nextStateFieldsObject,
-      __FIELD_CHANGED__: id,
-      __FORM_STATE__: getUpdatedFormState(json.form, nextStateFields)
-    })
-
-    if (typeof fieldsToUpdate === 'object' && fieldsToUpdate !== null)
-      Object.entries(fieldsToUpdate).forEach(async ([key, value]) => {
-        await handlerChange(key, value)
+      clearTimeout(timerShowSpinner)
+      setStateFields(nextStateFields)
+      const nextStateFieldsObject = useNativeFieldType
+        ? fieldsToObjectNativeTypes(nextStateFields)
+        : fieldsToObject(nextStateFields)
+
+      const fieldsToUpdate = await onChange({
+        ...nextStateFieldsObject,
+        __AUTOCOMPLETED_FIELD__: Boolean(autocompletedField),
+        __FIELD_CHANGED__: id,
+        __FORM_STATE__: getUpdatedFormState(json.form, nextStateFields)
       })
 
-    setStateShowSpinner(false)
+      if (typeof fieldsToUpdate === 'object' && fieldsToUpdate !== null) {
+        const autocompletedFields = true
+        Object.entries(fieldsToUpdate).forEach(async ([key, value]) => {
+          await handlerChange(key, value, undefined, autocompletedFields)
+        })
```

Comment:
If exists a response on `onChange`, we can assume that a field will be autocomplete, so we call `handlerChange` with that param with a `true` value.